src/addons/blog/services/blog-sync.ts

```diff
@@ -0,0 +1,254 @@
+ export class AddonBlogSyncProvider extends CoreSyncBaseProvider<AddonBlogSyncResult> {
+
+    protected componentTranslatableString = 'addon.blog.blog';
+
+    constructor() {
+        super('AddonBlogSyncService');
+    }
+
+     /**
+      * Try to synchronize all the entries in a certain site or in all sites.
+      *
+      * @param siteId Site ID to sync. If not defined, sync all sites.
+      * @param force Force sync.
+      * @returns Promise resolved if sync is successful, rejected if sync fails.
+      */
+    async syncAllEntries(siteId?: string, force?: boolean): Promise<void> {
+        await this.syncOnSites('All entries', (siteId) => this.syncAllEntriesFunc(siteId, !!force), siteId);
+    }
+
+    /**
+     * Sync all entries on a site.
+     *
+     * @param siteId Site ID to sync.
+     * @param force Force sync.
+     * @returns Promise resolved if sync is successful, rejected if sync fails.
+     */
+    protected async syncAllEntriesFunc(siteId?: string, force = false): Promise<CoreSyncResult> {
+        return await this.syncEntriesForSite(siteId || CoreSites.getCurrentSiteId(), force);
+    }
+
+    /**
+     * Perform entries syncronization for specified site.
+     *
+     * @param siteId Site id.
+     * @returns Syncronization result.
+     */
+    async syncEntriesForSite(siteId?: string, force: boolean = false): Promise<CoreSyncResult> {
+        const currentSyncPromise = this.getOngoingSync(ADDON_BLOG_SYNC_ID, siteId);
+
+        if (currentSyncPromise) {
+            return currentSyncPromise;
+        }
+
+        const needed = await this.isSyncNeeded(ADDON_BLOG_SYNC_ID, siteId);
+
+        if (force && !needed) {
+            return { updated: false, warnings: [] };
+        }
+
+        const entriesToSync = await this.getPendingToSyncEntries(siteId);
+
+        if (!entriesToSync.length) {
+            return { updated: false, warnings: [] };
+        }
+
+        this.logger.debug('Try to sync ' + ADDON_BLOG_SYNC_ID + ' in site ' + siteId);
+
+        const syncronizations = await Promise.all(entriesToSync.map(async (entry): Promise<CoreSyncResult> => {
+            if (CoreSync.isBlocked(AddonBlogProvider.COMPONENT, entry.created, siteId)) {
+                this.logger.debug('Cannot sync entry ' + entry.created + ' because it is blocked.');
+
+                throw new CoreSyncBlockedError(Translate.instant('core.errorsyncblocked', { $a: this.componentTranslate }));
+            }
+
+            const formattedEntry: AddonBlogAddEntryWSParams & { created: number } = {
+                subject: entry.subject,
+                summary: entry.summary,
+                summaryformat: entry.summaryformat,
+                options: JSON.parse(entry.options),
+                created: entry.created,
+            };
+
+            try {
+                if (entry.id) {
+                    await this.syncUpdatedEntry({ ...entry, id: entry.id, options: JSON.parse(entry.options) });
```

Comment:
```suggestion
                    await this.syncUpdatedEntry({ ...entry, id: entry.id, options: formattedEntry.options });
```
You already parsed the options so you can reuse them instead of parsing them again, right?